pkg/ee/validation/machine/providers.go

```diff
@@ -45,6 +46,38 @@ import (
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func GetMachineResourceUsage(ctx context.Context, userClient ctrlruntimeclient.Client, machine *clusterv1alpha1.Machine,
+	caBundle *certificates.CABundle) (*ResourceDetails, error) {
+	config, err := types.GetConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read machine.spec.providerSpec: %w", err)
+	}
+
+	var quotaReq *ResourceDetails
+	// TODO add all providers
+	switch config.CloudProvider {
+	case types.CloudProviderFake:
+		quotaReq, err = getFakeQuotaRequest(config)
+	case types.CloudProviderAWS:
+		quotaReq, err = getAWSResourceRequirements(ctx, userClient, config)
```

Comment:
nit: `usage` or `allocation` makes more sense to me here (not a strong opinion).

```suggestion
	case types.CloudProviderAWS:
		quotaUsage, err = getAWSResourceUsage(ctx, userClient, config)
```
or
```suggestion
	case types.CloudProviderAWS:
		allocatedQuota, err = getAllocatedAWSResources(ctx, userClient, config)
```